tests/models/test_models.py

```diff
@@ -145,7 +145,11 @@ def test_optimizer_update(self):
         # check new_param is in optimizer
         # check old_param is NOT in optimizer
         p_new = torch.nn.Parameter(torch.zeros(10, 10))
-        optimized = update_optimizer(optimizer, {"new_param": p_new}, {"old_param": p})
+
+        # Here we cannot know what parameter group but there is only one so it should work
+        new_parameters = {"new_param": p_new}
+        new_parameters.update(dict(model.named_parameters()))
+        optimized = update_optimizer(optimizer, new_parameters, {"old_param": p})
```

Comment:
The old behaviour of update_optimizer was allowing the use of new_parameters dict that is not in the optimizer. This is problematic since the structure I introduce needs to know where are the existing parameters in the hierarchy in order to assign a group to the new parameters. For now I just had to change this test. I don't think the usage is problematic in general because update_optimizer is always used on model.named_parameters() anyway. I can precise that the new behaviour requires parameters from the current optimizer that will stay in the optimizer to be inside this new_parameters. But I don't think we really expect people to use update_optimizer out of scope anyways.

Reply:
This is fine, I would expect some breaking changes in the handling of parameter groups. Up to now we gave no guarantees about the anyway.